src/app/components/Ionic/IonicProvider.tsx

```diff
@@ -21,10 +26,25 @@ const IonicContextProvider: FC<PropsWithChildren> = ({ children }) => {
       }
     })
 
+    // TODO: appStateChange triggers on the web platform as well, using visibilitychange listener, consider reusing the code(downside @capacitor/app dependency on web & extension)
```

Comment:
It's either all or none platforms imo and this PR introduce different locking behavior for mobile.

- if we want to rely on capacitor in all platforms than I am not sure this logic belongs to this provider.
- general thought, don't we need lock logic in saga? keep timer and reset it on every dispatched action?

Reply:
Well, the problem is, do we want to include `@capacitor` dependencies in web/extension platform? Until bluetooth functionality isn't enabled on web/ext, I would try to avoid that.

The thing is, as stated in PR description, once you minimize the app, the code execution will halt on mobile. Meaning that timers don't work anymore. The only way to do it, is with background services - and I am not even sure that, how you can invoke the main thread, without opening the app. Thats why I wasn't 100% sure, if we want to do it the "easy" way, or with background services, as that changes things on web/ext platforms - sure app business logic, can be easily extracted and reused, even if we choose to approach it differently from web/ext and mobile perspective.

Reply:
yeah by keeping the timer I meant storing lockTimestamp and refreshing it based on user actions and not rely on "visibilitychange" event only.

Reply:
I don't understand. On mobile there are no user actions, until app becomes active?

Reply:
Oh you mean, if user doesn't do any event in 30sec, then lock the wallet either way. Well, I would rather avoid capturing those events.

Reply:
This was discussed in sync meeting and we will deal with unification or web platforms in the future.